internal/controller/controllers/common.go

```diff
@@ -315,6 +315,30 @@ func machineNetworksEntriesToArray(entries []hiveext.MachineNetworkEntry) []*mod
 	}).([]*models.MachineNetwork)
 }
 
+func apiVipsArrayToStrings(vips []*models.APIVip) []string {
+	return funk.Map(vips, func(vip *models.APIVip) string {
```

Comment:
We have go 1.18 now we don't need funk anymore, we have https://github.com/samber/lo 

But you may keep using `funk` for consistency with the rest of this file if you want

Reply:
Thanks for the suggestion! I had a look and it seems like instead of `funk.Map` I would be doing

```
import "github.com/samber/lo"

lo.Map[int64, string]([]int64{1, 2, 3, 4}, func(x int64, index int) string {
    return strconv.FormatInt(x, 10)
})
// []string{"1", "2", "3", "4"}
```

so I'd still need to import additional library and syntax-wise it's not much more clear than `funk.Map`. If there is no other push towards moving out of `funk`, I'd prefer to stay with it

Reply:
Actually I don't know why their README example is so ugly, if you use it properly it's simpler and gets rid of the cast at the end:

```go
func apiVipsArrayToStrings(vips []*models.APIVip) []string {
	return lo.Map(vips, func(vip *models.APIVip, _ int) string {
		return string(vip.IP)
	})
}
```

Not that performance matters here, but it's also much faster since it doesn't use ugly runtime `Interfaces{}` for everything